guides/source/ja/engines.md

```diff
@@ -1006,3 +1006,114 @@ module MyEngine
+
+## `on_load`フックを使用してコードを変更する方法
+
+コードを変更することは一般的に難しくないです。もし、`ActiveRecord::Base`みたいなRailsフレームワークを参照する一行のコードがあるとしたら、これを`on_load`フックで囲むことで実現できます。
+
+### 例題1
+
+```ruby
+ActiveRecord::Base.include(MyActiveRecordHelper)
+```
+
+これは次のように書けます。
+
+```ruby
+ActiveSupport.on_load(:active_record) { include MyActiveRecordHelper } # ここではselfはActiveRecord::Baseを指してますのでシンプルに`#include`を呼び出せます。
+```
+
+### 例題2
+
+```ruby
+ActionController::Base.prepend(MyActionControllerHelper)
+```
+
+これは次のように書けます。
+
+```ruby
+ActiveSupport.on_load(:action_controller_base) { prepend MyActionControllerHelper } # ここではselfはActiveRecord::Baseを指してますのでシンプルに`#prepend`を呼び出せます。
+```
+
+### 例題3
+
+```ruby
+ActiveRecord::Base.include_root_in_json = true
+```
+
+これは次のように書けます。
+
+```ruby
+ActiveSupport.on_load(:active_record) { self.include_root_in_json = true } # ここではselfはActiveRecord::Baseを指してます。
+```
+
+## 利用可能なフック
+
+これらはコードで使えるフックです。
+
+以下のクラスの初期化プロセスをフックしたい場合はこれらを利用することができます。
+
+| クラス                             | 利用可能なフック                      |
+| --------------------------------- | ------------------------------------ |
+| `ActionCable`                     | `action_cable`                       |
+| `ActionController::API`           | `action_controller_api`              |
+| `ActionController::API`           | `action_controller`                  |
+| `ActionController::Base`          | `action_controller_base`             |
+| `ActionController::Base`          | `action_controller`                  |
+| `ActionController::TestCase`      | `action_controller_test_case`        |
+| `ActionDispatch::IntegrationTest` | `action_dispatch_integration_test`   |
+| `ActionMailer::Base`              | `action_mailer`                      |
+| `ActionMailer::TestCase`          | `action_mailer_test_case`            |
+| `ActionView::Base`                | `action_view`                        |
+| `ActionView::TestCase`            | `action_view_test_case`              |
+| `ActiveJob::Base`                 | `active_job`                         |
+| `ActiveJob::TestCase`             | `active_job_test_case`               |
+| `ActiveRecord::Base`              | `active_record`                      |
+| `ActiveSupport::TestCase`         | `active_support_test_case`           |
+| `i18n`                            | `i18n`                               |
+
+## 設定フック
+
+次は利用可能は設定フックです。これらは特定のフレームワークにフックしたりはしませんが、代わりにアプリケーション全体のコンテキストで実行されます。
+
+| フック                   | ユースケース                                                                              |
+| ---------------------- | ------------------------------------------------------------------------------------- |
+| `before_configuration` | 一番目の実行される設定可能なブロックです。イニシャライザより先に実行されます。              |
+| `before_initialize`    | 二番目の実行される設定可能なブロックです。フレームワークが初期化される前に呼び出されます。                |
+| `before_eager_load`    | 三番目の実行される設定可能なブロックです。`config.cache_classes`がfalseの場合は実行されません。 |
```

Comment:
Third configurable block to run. Does not run if `config.cache_classes` set to false.